core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -114,6 +116,9 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
         raf.setLength(roundedNewSize)
         mmap = raf.getChannel().map(FileChannel.MapMode.READ_WRITE, 0, roundedNewSize)
         _maxEntries = mmap.limit / entrySize
+        if(newlyCreated){
+          mmap.put(contentBuffer.asInstanceOf[ByteBuffer])
+        }
         mmap.position(position)
```

Comment:
Thanks for your comments.
# When the index file is lost and resize function is called, the mmap is assigned by the raf.getChannel().map(FileChannel.MapMode.READ_WRITE, 0, roundedNewSize).
Mmap will no content, is empty.
As above, we should cp the old content to the new mmap.